src/js/buildStandalone.js

```diff
@@ -19,56 +19,56 @@ clientStandaloneRequest.onreadystatechange = function() {
 		return;
 	}
 
 	var htmlString = standalone_HTML_String.concat("");
-	var title = "PuzzleScript Game";
-	if (state.metadata.title!==undefined) {
-		title=state.metadata.title;
-	}
-	var homepage = "https://www.puzzlescript.net";
-	if (state.metadata.homepage!==undefined) {
-		homepage=state.metadata.homepage;
-		if (!homepage.match(/^https?:\/\//)) {
-			homepage = "https://" + homepage;
-		}
-	}
-	var homepage_stripped = homepage.replace(/^https?:\/\//,'');
-
-	var background_color="black";
-	if ('background_color' in state.metadata) {
-		background_color=state.bgcolor;		
-	}
-	htmlString = htmlString.replace(/___BGCOLOR___/g,background_color);	
+	var title = state.metadata.title ? state.metadata.title : "PuzzleScript Game";
 
-	var text_color="lightblue";
-	if ('text_color' in state.metadata) {
-		text_color = state.fgcolor;	
+	var homepage = state.metadata.homepage ? state.metadata.homepage : "https://www.puzzlescript.net";
+	if (!homepage.match(/^https?:\/\//)) {
+		homepage = "https://" + homepage;
 	}
-	htmlString = htmlString.replace(/___TEXTCOLOR___/g,text_color);	
-
-	htmlString = htmlString.replace(/__GAMETITLE__/g,title);
-
+	var homepage_stripped = homepage.replace(/^https?:\/\//,'');
 
-	htmlString = htmlString.replace(/__HOMEPAGE__/g,homepage);
-	htmlString = htmlString.replace(/__HOMEPAGE_STRIPPED_PROTOCOL__/g,homepage_stripped);
+	var background_color = ('background_color' in state.metadata) ? state.bgcolor : "black";
+	htmlString = htmlString.replace(/___BGCOLOR___/g, background_color);	
 
-	// $ has special meaning to JavaScript's String.replace 
-	// c.f.	https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_a_parameter
-	// basically: '$$'s are inserted as single '$'s.
+	var text_color = ('text_color' in state.metadata) ? state.fgcolor : "lightblue";
+	htmlString = htmlString.replace(/___TEXTCOLOR___/g, text_color);	
 
-	// First we double all strings - remember that replace interprets '$$' 
-	// as a single'$', so we need to type four to double
-	sourceCode = sourceCode.replace(/\$/g, '$$$$');
+	htmlString = htmlString.replace(/__GAMETITLE__/g, safeUser(title));
+	htmlString = htmlString.replace(/__HOMEPAGE__/g, safeUser(homepage));
+	htmlString = htmlString.replace(/__HOMEPAGE_STRIPPED_PROTOCOL__/g, safeUser(homepage_stripped));
 
-	// Then when we substitute them, the doubled $'s will be reduced to single ones.
-	htmlString = htmlString.replace(/"__GAMEDAT__"/g,sourceCode);
+	// $ has special meaning, so replace it by \v, then switch it back
+	htmlString = htmlString.replace(/"__GAMEDAT__"/, safeDollar(sourceCode));
+	htmlString = htmlString.replace(/\v/g, '$$');
 
+	// remove bad Windows chars
+	var fn = title.replace(/[<>:|*?]/g, '!').trim() + ".html";
 	var BB = get_blob();
 	var blob = new BB([htmlString], {type: "text/plain;charset=utf-8"});
-	saveAs(blob, title+".html");
+	saveAs(blob, fn);
```

Comment:
IMO we could also replace the comment here with a more descriptive name than `fn`, like `removeBadWindowsCharsFn`

Reply:
I agree. But it's not a function, so maybe `filename`?